### PR TITLE
Refactor ToneAlarm class(es) to inherit from ModuleBase

### PR DESCRIPTION
**Pull request by mcsauder**
Hi there,

This PR condenses the four ToneAlarm class implementations into an implementation that inherits from ModuleBase rather than CDev, relocates board specific tone_alarm.cpp contents to header files, renames files to hardware specific names, renames TONEALARM0_DEVICE_PATH to TONE_ALARM0_DEVICE_PATH, and attempts to organize/standardize/alphabetize #defines.

**Test data / coverage**
This PR is straightfoward to validate on the bench at startup and has been bench tested on pixracer (stm32 fmu-v4) and pixhawk 4 (stm32 fmu-v5).  Audible tones are unchanged by this refactor work.  Hardware testing of the kinetis and samv7 builds has not yet been validated.

**Describe problem solved by the proposed pull request**
This PR standardizes the tone_alarm driver to inherit from the ModuleBase class and reduces the number of instances of the ToneAlarm class implementations to a primary class that is shared/built by respective hardware instances.

Currently tone_alarm.cpp exists in 4 incarnations in the src/drivers directory, (kinetis, samv7, stm32, and tone_alarm_sim).  Most of the code across files is copy/paste with only minor differences for hardware specific cases.  This PR refactors methods to allow common functions to be utilized to the greatest extent possible.

**Describe your preferred solution**
As a short-term solution that functions and reduces code duplication, this PR is a first functional step to inherit from the ModuleBase class.

However, additional work after this PR is likely warranted to create an inheritance hierarchy so that each hardware driver class can inherit from a ToneAlarm base class that inherits from ModuleBase.

**Describe possible alternatives**
Establishing an inheritance hierarchy should be pursued, (however a working solution to that design pattern has so far eluded me).

Please let me know if you have questions/comments/concerns on this PR.  Thanks!

-Mark

### CODE REVIEW

File: src/lib/tunes/tunes.h [hunk 1]
```diff
@@ -210,4 +185,32 @@ class Tunes
...
+	static const unsigned int _default_tunes_size;
+
+	const char *_tune = nullptr; ///< current tune string
+	const char *_next = nullptr; ///< next note in the string
+	const char *_tune_start_ptr = nullptr; ///< pointer to repeat tune
+
+	int _current_tune_id = static_cast<int>(TuneID::NONE);
+	bool _repeat = false;	     ///< if true, tune restarts at end
+
+	unsigned _default_note_length;
```
Comment (dagar) [inline src/lib/tunes/tunes.h]: Initialize all fields?

File: src/drivers/samv7/tone_alarm/CMakeLists.txt [hunk 2]
```diff
@@ -35,7 +35,8 @@ px4_add_module(
 	MODULE drivers__tone_alarm
 	MAIN tone_alarm
 	SRCS
-		tone_alarm.cpp
+		tone_alarm_samv7.cpp
+		../../tone_alarm/tone_alarm.cpp
```
Comment (dagar) [inline src/drivers/samv7/tone_alarm/CMakeLists.txt]: We should do this by creating separate libraries in each folder and linking.
To link all you have to do is add the library in the DEPENDS of a px4 module.

**Comment by dagar** [comment]
Thanks Mark, I'll need to find some time to review this.

File: src/lib/tunes/tunes.h (hunk 1 above)
Comment (mcsauder) [reply src/lib/tunes/tunes.h]: Thanks Daniel!  Done and pushed up.

**Comment by mcsauder** [comment]
Thanks for looking at this PR @dagar , I realize it's a painful review.  The purpose of this PR is to move toward a design pattern that can be replicated for other drivers that share similar copy/paste code duplication.  If ModuleBase inheritance is appropriate for most drivers and if an inheritance hierarchy can be established here, it lays the groundwork for other drivers that require slight tweaks for specific boards but not a whole new implementation from scratch for each instance.  Thanks again for your suggestions.

**Comment by mcsauder** [comment]
I'm changing my opinion on ModuleBase as a base class for drivers.  It appears to be unsuitable for establishing an inheritance hierarchy so that multiple classes of one type of driver can be created from a base class of that type of driver as is the case with tone_alarm in this PR as well as for IMU drivers.

As a result, I think I'll attempt to rewrite this PR utilizing the original CDev inheritance, not ModuleBase.

**Pull request closed without merging**